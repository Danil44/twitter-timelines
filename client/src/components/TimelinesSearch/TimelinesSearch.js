import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Toolbar } from '@material-ui/core';
import 'antd/dist/antd.css';
import { Input } from 'antd';
import { withRouter } from 'react-router-dom';
import styles from './TimelinesSearch.module.css';

const { Search } = Input;

export class TimelinesSearch extends Component {
  state = { username: '' };

  static propTypes = {
    onSubmit: PropTypes.func.isRequired,
    history: PropTypes.shape({
      replace: PropTypes.func.isRequired,
    }).isRequired,
  };

  handleChange = ({ target }) => {
    const { value, name } = target;

    this.setState({ [name]: value });
  };

  handleSubmit = () => {
    const { username } = this.state;
    const { onSubmit, history } = this.props;

    history.replace({
      pathname: '/search',
      search: `query=${username}`,
    });

    onSubmit(username);

    this.setState({ username: '' });
  };

  render() {
    const { username } = this.state;
    return (
      <Toolbar>
        <Search
          className={styles.input}
          name="username"
          placeholder="Search..."
          value={username}
          onChange={this.handleChange}
          onSearch={this.handleSubmit}
          style={{ maxWidth: 320 }}
          enterButton
        />
      </Toolbar>
    );
  }
}

export default withRouter(TimelinesSearch);
