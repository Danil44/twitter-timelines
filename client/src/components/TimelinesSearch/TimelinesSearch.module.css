@import '~antd/dist/antd.css';

.input {
  margin: 0 0 0 auto;
}
