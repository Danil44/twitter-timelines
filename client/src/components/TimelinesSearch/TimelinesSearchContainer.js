import { connect } from 'react-redux';
import { getTimelinesStart } from '../../redux/timelines/timelinesActions';
import TimelinesSearch from './TimelinesSearch';

const mapDispatchToProps = dispatch => ({
  onSubmit: value => dispatch(getTimelinesStart(value)),
});

export default connect(
  null,
  mapDispatchToProps,
)(TimelinesSearch);
