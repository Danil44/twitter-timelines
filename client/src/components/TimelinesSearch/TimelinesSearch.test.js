import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import { TimelinesSearch } from './TimelinesSearch';

describe('<TimelinesSearch/>', () => {
  let wrapper;

  const props = {
    onSubmit: jest.fn(),
    history: { replace: jest.fn() },
  };

  beforeEach(() => {
    wrapper = mount(<TimelinesSearch {...props} />);
  });

  it('should renders without crashing given the required props', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should call handleSubmit method which calls onSubmit prop', () => {
    wrapper.instance().handleSubmit();

    expect(props.onSubmit.mock.calls.length).toBe(1);
  });
});
