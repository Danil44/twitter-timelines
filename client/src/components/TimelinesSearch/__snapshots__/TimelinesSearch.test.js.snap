// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TimelinesSearch/> should renders without crashing given the required props 1`] = `
<TimelinesSearch
  history={
    Object {
      "replace": [MockFunction],
    }
  }
  onSubmit={[MockFunction]}
>
  <WithStyles(ForwardRef(Toolbar))>
    <ForwardRef(Toolbar)
      classes={
        Object {
          "dense": "MuiToolbar-dense",
          "gutters": "MuiToolbar-gutters",
          "regular": "MuiToolbar-regular",
          "root": "MuiToolbar-root",
        }
      }
    >
      <div
        className="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
      >
        <Search
          className="input"
          enterButton={true}
          name="username"
          onChange={[Function]}
          onSearch={[Function]}
          placeholder="Search..."
          style={
            Object {
              "maxWidth": 320,
            }
          }
          value=""
        >
          <Input
            addonAfter={
              <Button
                block={false}
                className="ant-input-search-button"
                ghost={false}
                htmlType="button"
                loading={false}
                onClick={[Function]}
                type="primary"
              >
                <Icon
                  type="search"
                />
              </Button>
            }
            className="ant-input-search input ant-input-search-enter-button"
            disabled={false}
            name="username"
            onChange={[Function]}
            onPressEnter={[Function]}
            placeholder="Search..."
            prefixCls="ant-input"
            style={
              Object {
                "maxWidth": 320,
              }
            }
            type="text"
            value=""
          >
            <span
              className="ant-input-search input ant-input-search-enter-button ant-input-group-wrapper"
              style={
                Object {
                  "maxWidth": 320,
                }
              }
            >
              <span
                className="ant-input-wrapper ant-input-group"
              >
                <input
                  className="ant-input"
                  disabled={false}
                  name="username"
                  onChange={[Function]}
                  onKeyDown={[Function]}
                  placeholder="Search..."
                  style={null}
                  type="text"
                  value=""
                />
                <span
                  className="ant-input-group-addon"
                >
                  <Button
                    block={false}
                    className="ant-input-search-button"
                    ghost={false}
                    htmlType="button"
                    key="enterButton"
                    loading={false}
                    onClick={[Function]}
                    type="primary"
                  >
                    <Wave>
                      <button
                        className="ant-btn ant-input-search-button ant-btn-primary"
                        onClick={[Function]}
                        type="button"
                      >
                        <Icon
                          key=".0"
                          type="search"
                        >
                          <LocaleReceiver
                            componentName="Icon"
                          >
                            <i
                              aria-label="icon: search"
                              className="anticon anticon-search"
                            >
                              <IconReact
                                className=""
                                type="search-o"
                              >
                                <svg
                                  aria-hidden="true"
                                  className=""
                                  data-icon="search"
                                  fill="currentColor"
                                  focusable="false"
                                  height="1em"
                                  key="svg-search"
                                  viewBox="64 64 896 896"
                                  width="1em"
                                >
                                  <path
                                    d="M909.6 854.5L649.9 594.8C690.2 542.7 712 479 712 412c0-80.2-31.3-155.4-87.9-212.1-56.6-56.7-132-87.9-212.1-87.9s-155.5 31.3-212.1 87.9C143.2 256.5 112 331.8 112 412c0 80.1 31.3 155.5 87.9 212.1C256.5 680.8 331.8 712 412 712c67 0 130.6-21.8 182.7-62l259.7 259.6a8.2 8.2 0 0 0 11.6 0l43.6-43.5a8.2 8.2 0 0 0 0-11.6zM570.4 570.4C528 612.7 471.8 636 412 636s-116-23.3-158.4-65.6C211.3 528 188 471.8 188 412s23.3-116.1 65.6-158.4C296 211.3 352.2 188 412 188s116.1 23.2 158.4 65.6S636 352.2 636 412s-23.3 116.1-65.6 158.4z"
                                    key="svg-search-svg-0"
                                  />
                                </svg>
                              </IconReact>
                            </i>
                          </LocaleReceiver>
                        </Icon>
                      </button>
                    </Wave>
                  </Button>
                </span>
              </span>
            </span>
          </Input>
        </Search>
      </div>
    </ForwardRef(Toolbar)>
  </WithStyles(ForwardRef(Toolbar))>
</TimelinesSearch>
`;
